processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -409,14 +420,28 @@ public String apply(@Nullable String input)
           @Nullable ArrayList<Iterable<Rowboat>> boats
       )
       {
-        return CombiningIterable.create(
-            new MergeIterable<Rowboat>(
-                Ordering.<Rowboat>natural().nullsFirst(),
-                boats
-            ),
-            Ordering.<Rowboat>natural().nullsFirst(),
-            new RowboatMergeFunction(sortedMetricAggs)
-        );
+        if (rollup) {
+          return CombiningIterable.create(
+              new MergeIterable<Rowboat>(
+                  Ordering.<Rowboat>natural().nullsFirst(),
+                  boats
+              ),
+              Ordering.<Rowboat>natural().nullsFirst(),
+              new RowboatMergeFunction(sortedMetricAggs)
+          );
+        } else {
+          return new MergeIterable<Rowboat>(
+              new Ordering<Rowboat>()
+              {
+                @Override
+                public int compare(Rowboat left, Rowboat right)
+                {
+                  return Longs.compare(left.getTimestamp(), right.getTimestamp());
```

Comment:
If a well-defined ordering is really needed, maybe the ConcurrentLinkedQueues in the PlainFactsHolder map could be changed to priority queues,  but this seems fine for now if there are no problems with queries.

Can you add the new no-rollup indexes from `TestIndex` to the `makeQueryRunners` in `QueryRunnerTestHelper` and check that all of the query unit tests pass?